app/services/pages/loader.rb

```diff
@@ -0,0 +1,12 @@
+module Pages
+  class Loader
+    def self.load
+      Pages::Manager.add_page({ :controller => :smart_proxies, :action => :show }, "smart_proxies/show") do |page|
```

Comment:
This is not extension but registration of existing pages and tabs. It is executed in the main app Rails initializer.

Plugins do have `extend*` methods to hook up on these. Similar approach can be seen in Dashboard Widgets.